tls/s2n_client_hello.c

```diff
@@ -604,8 +604,17 @@ int s2n_client_hello_send(struct s2n_connection *conn)
         POSIX_GUARD(s2n_stuffer_write_bytes(out, cipher->iana_value, S2N_TLS_CIPHER_SUITE_LEN));
     }
 
+    /**
+     *= https://tools.ietf.org/rfc/rfc5746#3.4
+     *# o  The client MUST include either an empty "renegotiation_info"
+     *#    extension, or the TLS_EMPTY_RENEGOTIATION_INFO_SCSV signaling
+     *#    cipher suite value in the ClientHello.  Including both is NOT
+     *#    RECOMMENDED.
+     *
+     * For maximum backwards compatibility, we choose to use the TLS_EMPTY_RENEGOTIATION_INFO_SCSV cipher suite
+     * to signal support for secure renegotiation rather than the "renegotiation_info" extension.
```

Comment:
I mean, we don't support secure renegotiation right?

Reply:
Oh the wording there is REALLY tricky. Maybe you can suggest a better way to discuss it.

The attack against renegotiation that prompted the "secure renegotiation" RFC involves making a renegotiation handshake look like an initial handshake. That attack is possible whether or not renegotiation is supported, because the legitimate client and server involved don't think they're doing renegotiation. So even if you don't support renegotiation, you have to support the part of "secure renegotiation" that advertises that you don't support renegotiation.

Reply:
"For maximum backwards compatibility, we choose to use the TLS_EMPTY_RENEGOTIATION_INFO_SCSV cipher suite rather than the "renegotiation_info" extension. This prevents us from advertising support for insecure renegotiation."

Reply:
Discussed offline: the solution is just removing that line completely, since it's not really important anyway :)